frontend/lib/src/features/user_profile/profile_info.dart

```diff
@@ -17,100 +25,181 @@ class ProfileInfo extends StatelessWidget {
-                    name: 'address',
-                    label: 'Address',
-                  ),
-                  const SizedBox(height: 26),
-                  const DatePickerField(
-                    name: 'birthday',
-                    labelText: 'Birthday',
-                  ),
-                  Padding(
-                    padding: const EdgeInsets.only(top: 33),
-                    child: Row(
-                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
-                      children: [
-                        TextButton(
-                          onPressed: () {
-                            // TODO: Implement using Getx
-                            profileAppNav.currentState
-                                ?.pushNamed('/profileChangePassword');
-                          },
-                          style: TextButton.styleFrom(
-                            padding: const EdgeInsets.all(0),
-                          ),
-                          child: Text(
-                            'Change Password',
-                            style: Theme.of(context)
-                                .textTheme
-                                .apply(bodyColor: const Color(0xFF00A3FF))
-                                .labelSmall,
-                          ),
-                        ),
-                        Button(
-                          text: 'Save',
-                          padding:
-                              const EdgeInsets.symmetric(vertical: 16, horizontal: 50),
-                          onPressed: () {
-                            if (formKey.currentState?.saveAndValidate() ?? false) {
-                              debugPrint(formKey.currentState?.value.toString());
-                            }
-                          },
+    UserProfileController controller = Get.put(UserProfileController());
+
+    onSubmit() async {
+      controller.setLoading = true;
+      final userId = await storage.read(key: StorageKeys.userId.name);
+
+      final editProfile = await NetworkConfig().client.put(
+        '$usersUrl/$userId',
+        data: {
+          'first_name': controller.firstName,
+          'middle_name': controller.middleName,
+          'last_name': controller.lastName,
+          'address': controller.address,
+          'birthday': controller.birthday.toString(),
+        },
+      );
+
+      controller.setLoading = false;
+
+      if (editProfile.statusCode == HttpStatus.ok) {
+        showSnackbar(
+          title: "Success",
+          content: "Profile updated successfully.",
+        );
+      } else if (editProfile.statusCode == HttpStatus.badRequest) {
+        showSnackbar(
+          title: "Failed",
+          content: "Missing field",
+        );
+      }
+    }
+
+    return GetX<UserProfileController>(
+      builder: (profile) {
+        return profile.isLoading
+            ? const CircularProgressIndicator()
+            : GestureDetector(
```

Comment:
refactor
avoid ternary expression for large code blocks


if isloading {
  return circ
}

return gesture